src/Microsoft.AspNet.Mvc.Core/Rendering/HtmlHelperInputExtensions.cs

```diff
@@ -16,26 +16,27 @@ public static class HtmlHelperInputExtensions
         /// "hidden" with value "false".
         /// </summary>
         /// <param name="htmlHelper">The <see cref="IHtmlHelper"/> instance this method extends.</param>
-        /// <param name="name">Expression name, relative to the current model.</param>
+        /// <param name="expression">Expression name, relative to the current model.</param>
         /// <returns>A new <see cref="HtmlString"/> containing the &lt;input&gt; elements.</returns>
         /// <remarks>
         /// <para>
-        /// Combines <see cref="TemplateInfo.HtmlFieldPrefix"/> and <paramref name="name"/> to set checkbox
-        /// element's "name" attribute. Sanitizes <paramref name="name"/> to set checkbox element's "id" attribute.
+        /// Combines <see cref="TemplateInfo.HtmlFieldPrefix"/> and <paramref name="expression"/> to set checkbox
+        /// element's "name" attribute. Sanitizes <paramref name="expression"/> to set checkbox element's "id"
+        /// attribute.
         /// </para>
         /// <para>Determines checkbox element's "checked" attribute based on the following precedence:</para>
         /// <list type="number">
         /// <item>
-        /// <see cref="ModelBinding.ModelStateDictionary"/> entry for <paramref name="name"/> (converted to a
+        /// <see cref="ModelBinding.ModelStateDictionary"/> entry for <paramref name="expression"/> (converted to a
         /// fully-qualified name) if entry exists and can be converted to a <see cref="bool"/>.
         /// </item>
         /// <item>
-        /// <see cref="ViewDataDictionary"/> entry for <paramref name="name"/> (converted to a fully-qualified name)
-        /// if entry exists and can be converted to a <see cref="bool"/>.
+        /// <see cref="ViewDataDictionary"/> entry for <paramref name="expression"/> (converted to a fully-qualified
+        /// name) if entry exists and can be converted to a <see cref="bool"/>.
```

Comment:
:-1: `expression` names are relative to the current model.  this is about generating an absolute name i.e. one that's relative to whatever the top-level `Model` is.  result is normally a unique name withing the generated page.  and we generate a "fully-qualified name" from lambda `expression` parameters as well.